pkg/karmadactl/cmdinit/kubernetes/deploy.go

```diff
@@ -170,12 +170,21 @@ func (i *CommandInitOption) Complete() error {
 		}
 	}
 
-	// Determine whether KarmadaDataPath exists, if so, delete it
-	if utils.IsExist(i.KarmadaDataPath) {
-		if err := os.RemoveAll(i.KarmadaDataPath); err != nil {
+	return initializeDirectory(i.KarmadaDataPath)
+}
+
+// initializeDirectory clean up the directory path if it exists, then recreate it
+func initializeDirectory(path string) error {
+	// Initialize karmada-data directory.
```

Comment:
It's ok to introduce a function here, but the comment might be inappropriate, as it's not dedicated to `karmada-data` now. It's a `common` function now.

Reply:
I guess we can add a unit test for it.

Reply:
fixed, and add ut case